### ISSUE TITLE
QIE mapping

### ISSUE THREAD

**Issue by awhitbeck** [elliot-hughes/hcal_teststand_scripts#2]
QIE mapping is too slow... need to do some research on how to speed it up.  Some ideas:

Use the uhtr histograms to read out data - which is very fast.

Use the link pattern readout to first figure out where groups of four QIEs then set those groups of four to either fixed range mode 0,1,2,3.  

-the combination of the two should be very fast

@Teststandees/citizens

**Comment by awhitbeck** [comment]
did we ever debug the faster mapping code?

**Comment by elliot-hughes** [comment]
No, but now's a good time to do this, since the map changed. I'll do it.

**Comment by elliot-hughes** [comment]
This is more annoying than I anticipated. When I use the slow version, I get results like

```
{"slot": 2, "uhtr_slot": 11, "fiber": 1, "link": 16, "half": 0, "qie": 17, "crate": 1, "id": "0x00000BAD 0x00000000", "channel": 3},
```

The `"id"` field is odd. I'm going to go to lunch and look into this after.

**Comment by awhitbeck** [comment]
How are you reading the unique ID?  From the device itself or from the IGLOO2 register?

**Comment by elliot-hughes** [comment]
The `"0x00000BAD 0x00000000"` unique ID was due to a bug introduced by removing quotes from the returned results of `ngccm.send_commands_parsed`. I [fixed the bug](https://github.com/Teststandees/hcal_teststand_scripts/blob/master/hcal_teststand/qie.py#L183-L184). I updated the [map in the repository](https://github.com/Teststandees/hcal_teststand_scripts/blob/master/configuration/maps/904_qie_map.json).

**Comment by awhitbeck** [comment]
nice!

**Comment by awhitbeck** [comment]
can this be closed?
